<a href="https://colab.research.google.com/github/ZozolNat/teor_veroiatnosti-i-stat/blob/main/%D0%B43_8_%D1%81%D1%82%D0%B0%D1%82_%D0%B8_%D0%B2%D0%B5%D1%80%D0%BE%D1%8F%D1%82.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#1) Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks): zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110], ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]. Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy. Полученные значения должны быть равны. Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков, а затем с использованием функций из библиотек numpy и pandas. 


описание и решение

 считаем ковариацию в ручную по формуле 

cov(X, Y) = (Σ((x_i - mean(X)) * (y_i - mean(Y)))) / (n), где X и Y - две случайные величины, x_i и y_i - их соответствующие значения, mean(X) и mean(Y) - средние значения, а n - количество наблюдений.

сначала надр найти средние значении как в предыдущих задачах ДЗ ( сумма значений/ на длину ( колличество значений)), получается 

mean_zp = (35 + 45 + 190 + 200 + 40 + 70 + 54 + 150 + 120 + 110) / 10 = 101.4

mean_ks = (401 + 574 + 874 + 919 + 459 + 739 + 653 + 902 + 746 + 832) / 10 = 709.9

далее считаем ковариацию по формуле указанной выше

cov = ((zp[0] - mean_zp) * (ks[0] - mean_ks) + 
       (zp[1] - mean_zp) * (ks[1] - mean_ks) + 
       (zp[2] - mean_zp) * (ks[2] - mean_ks) + 
       (zp[3] - mean_zp) * (ks[3] - mean_ks) + 
       (zp[4] - mean_zp) * (ks[4] - mean_ks) + 
       (zp[5] - mean_zp) * (ks[5] - mean_ks) + 
       (zp[6] - mean_zp) * (ks[6] - mean_ks) + 
       (zp[7] - mean_zp) * (ks[7] - mean_ks) + 
       (zp[8] - mean_zp) * (ks[8] - mean_ks) + 
       (zp[9] - mean_zp) * (ks[9] - mean_ks)) / (10 ) =
       = ((35 - mean_zp) * (401 - mean_ks) + 
       (45 - mean_zp) * (574 - mean_ks) + 
       (190 - mean_zp) * (874 - mean_ks) + 
       (200 - mean_zp) * (919 - mean_ks) + 
       (40 - mean_zp) * (459 - mean_ks) + 
       (70 - mean_zp) * (739 - mean_ks) + 
       (54 - mean_zp) * (653 - mean_ks) + 
       (150 - mean_zp) * (902 - mean_ks) + 
       (120 - mean_zp) * (746 - mean_ks) + 
       (110 - mean_zp) * (832 - mean_ks)) / (10) = (-7151.66 + 1454.94 + 2927.24 + 3427.96 - 6937.36 + 3593.76 + 1801.64 + 4545.26 + 2861.04 + 3160.76) / 10 = 9157.8



In [6]:
import numpy as np

zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110]
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]

mean_zp = np.mean(zp)
mean_ks = np.mean(ks)

n = len(zp)

cov = np.sum((zp - mean_zp) * (ks - mean_ks)) / n

print(cov)
# считаем в пайтон с учетом указанной формулы

9157.84


теперь с помощью функции cov из numpy

In [3]:
import numpy as np

zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110]
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]

cov = np.cov(zp, ks, ddof=0)[0, 1]
print("ковариация:", cov)


ковариация: 9157.84


Дале для вычисления коэффициента корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков можно используем формулу: corr = cov / (std_zp * std_ks), где cov - ковариация между признаками, std_zp - стандартное отклонение признака zp, std_ks - стандартное отклонение признака ks.

*ищем с использованием NumPy с:

import numpy as np

cov = np.cov(zp, ks, ddof=1)[0, 1]
std_zp = np.std(zp, ddof=1)
std_ks = np.std(ks, ddof=1)

corr = cov / (std_zp * std_ks)

print(corr)


**ищем с использованием Pandas:

import pandas as pd

df = pd.DataFrame({'zp': zp, 'ks': ks})

corr = df['zp'].corr(df['ks'])

print(corr)


In [8]:
import numpy as np

zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110]
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]

cov = np.cov(zp, ks, ddof=1)[0, 1]
std_zp = np.std(zp, ddof=1)
std_ks = np.std(ks, ddof=1)

corr = cov / (std_zp * std_ks)

print(corr)
#с использованием NumPy


0.887490092073916


In [11]:
import pandas as pd

df = pd.DataFrame({'zp': [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
                   'ks': [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]})

corr = df['zp'].corr(df['ks'])

print(corr)
#с использованием pandAS

0.8874900920739162


#2) Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:131, 125, 115, 122, 131, 115, 107, 99, 125, 111. Известно, что в генеральной совокупности IQ распределен нормально. Найдите доверительный интервал для математического ожидания с надежностью 0.95.


ОПИСАНИЕ И РЕШЕНИЕ 

Доверительный интервал = (среднее значение выборки) ± (критическое значение * стандартная ошибка),

где:

Среднее значение выборки - среднее арифметическое измерений в выборке,
Критическое значение - определяется с учетом надежности и размера выборки. Для надежности 0.95 и выборки размером 10 можно использовать критическое значение t-распределения с (n-1) степенями свободы. В данном случае n = 10, поэтому нам понадобится значение t-критерия для 9 степеней свободы,
Стандартная ошибка - стандартное отклонение выборки, деленное на квадратный корень из размера выборки.

*** СМОТРЕТЬ ЗАДАЧИ В ПРЕДЫДУЩИХ дз, АНАЛОГ.


пишем в пайтоне


In [12]:
import numpy as np
from scipy.stats import t

data = [131, 125, 115, 122, 131, 115, 107, 99, 125, 111]
confidence = 0.95

mean = np.mean(data)
std = np.std(data, ddof=1)
n = len(data)
critical_value = t.ppf((1 + confidence) / 2, df=n-1)
standard_error = std / np.sqrt(n)

lower_bound = mean - critical_value * standard_error
upper_bound = mean + critical_value * standard_error

confidence_interval = (lower_bound, upper_bound)

print("Доверительный интервал:", confidence_interval)


Доверительный интервал: (110.55608365158724, 125.64391634841274)


Т.о, с надежностью 0.95 можно сказать, что математическое ожидание IQ в генеральной совокупности находится в интервале от 110.56 до 125.64.

# 3) Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.


Описание и решение. СМОТРЕТЬ ЗАДАЧУ ВЫШЕ, АНАЛОГ.

Доверительный интервал = (среднее значение выборки) ± (критическое значение * стандартная ошибка),

где:

Среднее значение выборки - среднее арифметическое измерений в выборке,
Критическое значение - определяется с учетом надежности и размера выборки. Для надежности 0.95 и выборки размером 27 можно использовать критическое значение t-распределения с (n-1) степенями свободы. В данном случае n = 27, поэтому нам понадобится значение t-критерия для 26 степеней свободы,
Стандартная ошибка - стандартное отклонение выборки, деленное на квадратный корень из размера выборки.



In [13]:
import numpy as np
from scipy.stats import t

sample_mean = 174.2
sample_size = 27
population_variance = 25
confidence = 0.95

critical_value = t.ppf((1 + confidence) / 2, df=sample_size-1)
standard_error = np.sqrt(population_variance / sample_size)

lower_bound = sample_mean - critical_value * standard_error
upper_bound = sample_mean + critical_value * standard_error

confidence_interval = (lower_bound, upper_bound)

print("Доверительный интервал:", confidence_interval)


Доверительный интервал: (172.2220658754539, 176.17793412454608)


Т.о., с надежностью 0.95 можно сказать, что математическое ожидание роста футболистов в сборной находится в интервале от 172.22 до 176.17 см.